frontend/rust-lib/flowy-search/src/folder/indexer.rs

```diff
@@ -0,0 +1,360 @@
+use std::{any::Any, collections::HashMap, fs, path::Path, sync::Weak};
+
+use crate::{
+  entities::ResultIconTypePB,
+  folder::schema::{FolderSchema, FOLDER_ICON_FIELD_NAME, FOLDER_TITLE_FIELD_NAME},
+};
+use collab::core::collab::{IndexContent, IndexContentReceiver};
+use collab_folder::{View, ViewIcon, ViewIndexContent, ViewLayout};
+use flowy_error::{FlowyError, FlowyResult};
+use flowy_user::services::authenticate_user::AuthenticateUser;
+use lib_dispatch::prelude::af_spawn;
+use strsim::levenshtein;
+use tantivy::{
+  collector::TopDocs, directory::MmapDirectory, doc, query::QueryParser, Index, IndexReader,
+  IndexWriter, Term,
+};
+
+use crate::{
+  entities::SearchResultPB,
+  services::indexer::{IndexManager, IndexableData},
+};
+
+use super::{
+  entities::FolderIndexData,
+  schema::{FOLDER_ICON_TY_FIELD_NAME, FOLDER_ID_FIELD_NAME},
+};
+
+#[derive(Clone)]
+pub struct FolderIndexManager {
+  folder_schema: Option<FolderSchema>,
+  index: Option<Index>,
+  index_reader: Option<IndexReader>,
+}
+
+const FOLDER_INDEX_DIR: &str = "folder_index";
+
+impl FolderIndexManager {
+  pub fn new(auth_user: Weak<AuthenticateUser>) -> Self {
+    let authenticate_user = auth_user.upgrade();
```

Comment:
it would better if each step in the new function comes with simple documentation